dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/task/ParallelTaskGroup.java

```diff
@@ -42,21 +44,21 @@ public void addTask(Task<?> task) {
     super.addTask(task);
   }
 
-  private static class TaskRunner<T> implements Callable<T> {
+  private static class TaskRunner {
 
-    private final TaskRunContext context;
-    private final Task<T> task;
-    private final CSVPrinter printer;
+    @Nonnull private final TaskRunContext context;
+    @Nonnull private final Task<?> task;
+    @Nonnull private final CSVPrinter printer;
 
-    public TaskRunner(TaskRunContext context, Task<T> task, CSVPrinter printer) {
+    public TaskRunner(
+        @Nonnull TaskRunContext context, @Nonnull Task<?> task, @Nonnull CSVPrinter printer) {
       this.context = context;
       this.task = task;
       this.printer = printer;
     }
 
-    @Override
-    public T call() throws Exception {
-      T result = context.runChildTask(task);
+    public @Nullable Object call() throws IOException {
```

Comment:
It might not be Nullable. Whether it's Nullable or not can be captured from Task<T> by a type-use annotation on T, rather than being re-specified here.

Reply:
It's not a reflective call. It's a compiler type-system plugin.

Reply:
I assumed the intention was to get the annotation at run-time. At compile-time it seems like there's no information available, because the type parameter to the generic `TaskRunner` is always `?`.

Reply:
It _could_ be inspected at runtime, but I don't know if I've ever seen anyone do that. This is all about compile-time correctness, a shift-left of a runtime failure. That's why I'm sometimes a bit jumpy about these subtleties: Compile-time error detection is ideal, and I'd love to have the entire team understand more about how effective it can be.